src/perform_step/linear_perform_step.jl

```diff
@@ -23,7 +24,8 @@ function perform_step!(integrator, cache::MagnusMidpointCache, repeat_step = fal
             opnorm = integrator.opts.internalopnorm, iop = alg.iop)
     else
         A = Matrix(L) #size(L) == () ? convert(Number, L) : convert(AbstractMatrix, L)
-        u .= exp(dt * L) * u
+        cacheA = ExponentialUtilities.alloc_mem(A, exp_method)
```

Comment:
This cache should be constructed during the cache construction time.

Reply:
Sorry, I am a little confused, while the cache construction is located in ```linear_cache.jl```, you mean we should construct the exponential matrix cache during the algorithm cache construction process right? My understanding is that we can allocate the matrix once in algorithms cache construction like ```exp_cache = ExponentialUtilities.alloc_mem(f, exp_method)``` and pass the allocated matrix to perform_step and use copies of that allocated matrix to complete ```exponential!``` instead of allocating many times in perform_step.

Reply:
> My understanding is that we can allocate the matrix once in algorithms cache construction like exp_cache = ExponentialUtilities.alloc_mem(f, exp_method) and pass the allocated matrix to perform_step and use copies of that allocated matrix to complete exponential! instead of allocating many times in perform_step.

Yes exactly. The `alloc_mem` call is intended to be called just once, and its cache can be reused in every `exponential!` call from that point on. That would greatly reduce the allocations of these calls.